tools/extra/mmlink/main.cpp

```diff
@@ -142,26 +136,38 @@ int main( int argc, char** argv )
 	if ( argc < 2 ) {
 		MMLinkAppShowHelp();
 		return 1;
-	} else if ( 0 != (res = ParseCmds(&mmlinkCmdLine, argc, argv)) ) {
+	}
+
+	result = fpgaGetProperties(NULL, &filter);
+	ON_ERR_GOTO(result, out_exit, "creating properties object");
+
+	if (opae_set_properties_from_args(filter,
+					  &result,
+					  &argc,
+					  argv)) {
+		result = FPGA_EXCEPTION;
+		PRINT_ERR("parsing args.\n");
+		goto out_destroy_prop;
+	} else if (result) {
+		PRINT_ERR("setting properties from args.\n");
+		goto out_destroy_prop;
+	}
+
+	if ( 0 != (res = ParseCmds(&mmlinkCmdLine, argc, argv)) ) {
 		if (res != -2)
-			PRINT_ERR( "Error scanning command line \n.");
-		return 2;
+			PRINT_ERR("Error scanning command line.\n");
+		goto out_destroy_prop;
 	}
 
 	if ('\0' == mmlinkCmdLine.ip[0]) {
 		strncpy(mmlinkCmdLine.ip, "0.0.0.0", 8);
 		mmlinkCmdLine.ip[7] = '\0';
 	}
 
-	printf(" ------- Command line Input START ----\n\n");
-
-	printf(" Segment               : %d\n", mmlinkCmdLine.segment);
-	printf(" Bus                   : %d\n", mmlinkCmdLine.bus);
-	printf(" Device                : %d\n", mmlinkCmdLine.device);
-	printf(" Function              : %d\n", mmlinkCmdLine.function);
-	printf(" Socket-id             : %d\n", mmlinkCmdLine.socket);
-	printf(" Port                  : %d\n", mmlinkCmdLine.port);
-	printf(" IP address            : %s\n", mmlinkCmdLine.ip);
+	printf(" ------- Command line Input START ----\n");
+	printf(" Socket-id        : %d\n", mmlinkCmdLine.socket);
+	printf(" Port             : %d\n", mmlinkCmdLine.port);
+	printf(" IP address       : %s\n", mmlinkCmdLine.ip);
```

Comment:
I wonder if we really need to print this to stdout or make these like debug messages?

Reply:
Should be handled in a separate request, at any rate.